addons/point_of_sale/static/src/js/db.js

```diff
@@ -94,19 +94,16 @@ var PosDB = core.Class.extend({
                 name : 'Root',
             };
         }
-        for(var i=0, len = categories.length; i < len; i++){
-            this.category_by_id[categories[i].id] = categories[i];
-        }
-        len = categories.length;
-        for(i=0; i < len; i++){
-            var cat = categories[i];
-            var parent_id = cat.parent_id[0] || this.root_category_id;
-            this.category_parent[cat.id] = cat.parent_id[0];
-            if(!this.category_childs[parent_id]){
-                this.category_childs[parent_id] = [];
+        categories.forEach(function(cat){
+            self.category_by_id[cat.id] = cat;
+            var parent_id = (cat.parent_id[0] && categories.find(categ => categ.id == cat.parent_id[0])) ?
```

Comment:
Indeed, apparently PhantomJS doesn't support ES6 yet, I was not aware of this. I hope they will follow quickly, because some ES6 functionalities like promises are already used in Odoo dependencies.